spark3/src/main/java/org/apache/iceberg/spark/SparkSessionCatalog.java

```diff
@@ -242,6 +242,13 @@ public void renameTable(Identifier from, Identifier to) throws NoSuchTableExcept
 
   @Override
   public final void initialize(String name, CaseInsensitiveStringMap options) {
+    if (options.containsKey("type") && options.get("type").equalsIgnoreCase("hive") && options.containsKey("uri")) {
+      throw new UnsupportedOperationException("Cannot set an alternative uri when" +
+              " using the SparkSessionCatalog, " +
+              "make an alternative SparkCatalog if the " +
+              "Spark Session catalog and Iceberg " +
+              "catalog should contact different metastores.");
```

Comment:
We want to avoid long error messages. First, we should be more direct: "Cannot set uri for SparkSessionCatalog". Then we can add additional context, like "conflicts with Hive conf URI", if we don't think it is clear enough. Together, that's still a fairly short message: "Cannot set uri for SparkSessionCatalog: conflicts with Hive conf URI".

Reply:
done